test/testing/test_pytest_plugin.py

```diff
@@ -321,6 +321,41 @@ def test_container_factory(container_factory):
     assert result.ret == 0
 
 
+@config.patch({'FOO': 1})
+def test_container_factory_with_config_patch_argument():
+
+    from nameko import config
+    from nameko.testing.pytest import (
+        container_factory as container_factory_fixture
+    )
+
+    class Service(object):
+        name = "x"
+
+    assert config['FOO'] == 1
+
+    container_factory_fixture = container_factory_fixture()
+
+    container_factory = next(container_factory_fixture)
```

Comment:
I think it would be easier to understand these tests if they used the `testdir.runpytest` approach like the other tests in this module. Is there a reason not to do that here?